pkg/sqlcache/informer/listoption_indexer.go

```diff
@@ -323,28 +321,90 @@ func NewListOptionIndexer(ctx context.Context, s Store, opts ListOptionIndexerOp
 func (l *ListOptionIndexer) GetLatestResourceVersion() []string {
 	var latestRV []string
 
-	l.latestRVLock.RLock()
+	l.lock.RLock()
 	latestRV = []string{l.latestRV}
-	l.latestRVLock.RUnlock()
+	l.lock.RUnlock()
 
 	return latestRV
 }
 
+func watcherWithBackfill[T any](ctx context.Context, eventsChan chan<- T) (chan<- T, func()) {
+	backfill, backfillDone := context.WithCancel(ctx)
+	writeChan := make(chan T)
+	buffer := make(chan T, 100)
```

Comment:
nit: Should buffer be `buffer := make(chan []T, 1)` instead? That way we just fill up a list and after backfilling you pop from that list. (instead of involving channels).

It does mean that we lose on the "free" (but not quite) max length.

Reply:
Please see my comment below, explaining why using a channel instead of a slice, so we can keep the buffer size under control and temporarily block ingestion if that happens.